### PR TITLE
Home & Category Screens

### PR DESCRIPTION
**Pull request by roaabadawy**
- pages folder: Bottom navigation bar, Home Screen, Category Screen, profile Screen.

- widgets: my_Text, product_Card, Section header, search field.

- added color in constants folder.

- create assets folder and added images.

- packages: shared_preferences, dio, flutter_bloc, custom_navigation_bar, carousel_slider.

### CODE REVIEW

File: lib/resources/constants/app_constants.dart [hunk 1]
```diff
@@ -1,5 +1,32 @@
-class AppConstants {
+/*class AppConstants {
   AppConstants._();
 
-  
+
+}*/
+import 'package:flutter/material.dart';
+
```
Comment (ahmeddarafat) [inline lib/resources/constants/app_constants.dart:13]: - you can put colors in app_colors file

File: lib/resources/constants/app_constants.dart [hunk 2]
```diff
@@ -1,5 +1,32 @@
...
+Color kWhiteColor = const Color(0xFFE5E5E5);
+Color kLightGrayColor = const Color(0xFFF2F3F2);
+Color kGrayColor = const Color(0xFF7C7C7C);
+Color kDarkGrayColor = const Color(0xFF4C4F4D);
+
+ThemeData theme = ThemeData(
+  primaryColor: kPrimaryColor,
+  backgroundColor: kWhiteColor,
+);
+
```
Comment (ahmeddarafat) [inline lib/resources/constants/app_constants.dart:59]: - you can put images in app_assets file

File: lib/resources/constants/app_constants.dart [hunk 3]
```diff
@@ -1,5 +1,32 @@
...
+String getAsset(String image) => 'assets/images/$image.png';
+
```
Comment (ahmeddarafat) [inline lib/resources/constants/app_constants.dart:68]: - there are package called screen util I set up it in main.dart
- you just need to know how to use it

File: lib/view/pages/home_screen.dart [hunk 4]
```diff
@@ -0,0 +1,139 @@
+import 'package:flutter/material.dart';
+import 'package:carousel_slider/carousel_slider.dart';
+import 'package:iti_final_project/resources/constants/app_constants.dart';
+import '../widgets/my_text.dart';
+import '../widgets/product_card.dart';
+import '../widgets/search_field.dart';
+import '../widgets/section_header.dart';
+
```
Comment (ahmeddarafat) [inline lib/view/pages/home_screen.dart]: - you can put sliderImages list & popularCars list  & bestSeling list in app_constants.dart

File: lib/view/pages/home_screen.dart [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+                ),
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+}
+
```
Comment (ahmeddarafat) [inline lib/view/pages/home_screen.dart]: you can create file prodcut_model.dart and put it in data/model directory

File: lib/view/widgets/my_text.dart
Comment (ahmeddarafat) [inline lib/view/widgets/my_text.dart:1]: - I have created file called Public_Text in widgets folder, but you can use this or that which you want

File: lib/view/widgets/product_card.dart
Comment (ahmeddarafat) [inline lib/view/widgets/product_card.dart:1]: widgets folder is used for only public widgets  like text or button
but if there is a widget for specific page you can organize like that
view folder
---- pages folder
--------- home folder
-------------- components folder
------------------ product_card.dart file
-------------- home_page.dart file
--------- profile folder
-------------- components folder
------------------ profile_widget.dart file
-------------- profile_page.dart file
---- widget folder
.....

File: lib/resources/constants/app_constants.dart (hunk 1 above)
Comment (roaabadawy) [reply lib/resources/constants/app_constants.dart:13]: Done

File: lib/resources/constants/app_constants.dart (hunk 2 above)
Comment (roaabadawy) [reply lib/resources/constants/app_constants.dart:59]: Done

File: lib/view/pages/home_screen.dart (hunk 4 above)
Comment (roaabadawy) [reply lib/view/pages/home_screen.dart]: Done

File: lib/view/pages/home_screen.dart (hunk 5 above)
Comment (roaabadawy) [reply lib/view/pages/home_screen.dart]: Done

**Pull request merged**